From 856adde71a27982dff0da4de0e90aa9b5b0d7ce7 Mon Sep 17 00:00:00 2001
From: David Kilp <David.Kilp@crossmatch.com>
Date: Tue, 9 Oct 2018 14:39:24 -0400
Subject: [PATCH 03/18] On Android platforms set the name the battery guage
 registers into the /sys/class/power tree to the more generic "battery"
 instead of "bq27742-0" so that Android battery monitor apps (I'm looking at
 you 3C Battery Monitor) will see the battery.

---
 drivers/power/supply/bq27xxx_battery_i2c.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/drivers/power/supply/bq27xxx_battery_i2c.c b/drivers/power/supply/bq27xxx_battery_i2c.c
index 5c5c3a6..5541a67 100644
--- a/drivers/power/supply/bq27xxx_battery_i2c.c
+++ b/drivers/power/supply/bq27xxx_battery_i2c.c
@@ -83,7 +83,13 @@ static int bq27xxx_battery_i2c_probe(struct i2c_client *client,
 	if (num < 0)
 		return num;
 
+#ifdef CONFIG_ANDROID
+	// default name will show up as /sys/class/power_supply/bq27742-0
+	// Android battery app(s) need: /sys/class/power_supply/battery
+	name = devm_kasprintf(&client->dev, GFP_KERNEL, "%s", "battery");
+#else
 	name = devm_kasprintf(&client->dev, GFP_KERNEL, "%s-%d", id->name, num);
+#endif
 	if (!name)
 		goto err_mem;
 
-- 
2.9.5

